extensions/script/ExecuteScript.cpp

```diff
@@ -52,6 +52,28 @@ core::Property ExecuteScript::ModuleDirectory("Module Directory",  // NOLINT
 core::Relationship ExecuteScript::Success("success", "Script successes");  // NOLINT
 core::Relationship ExecuteScript::Failure("failure", "Script failures");  // NOLINT
 
+ScriptEngineFactory::ScriptEngineFactory(core::Relationship& success, core::Relationship& failure, std::shared_ptr<core::logging::Logger> logger)
+  : success_(success),
+    failure_(failure),
+    logger_(logger) {
+}
+
+ScriptEngineQueue::ScriptEngineQueue(uint8_t max_engine_count, ScriptEngineFactory& engine_factory, std::shared_ptr<core::logging::Logger> logger)
+    : max_engine_count_(max_engine_count),
+      engine_factory_(engine_factory),
+      logger_(logger) {
+}
+
+void ScriptEngineQueue::returnScriptEngine(std::shared_ptr<script::ScriptEngine>&& engine) {
+  const std::lock_guard<std::mutex> lock(queue_mutex_);
+  if (engine_queue_.size_approx() < max_engine_count_) {
+    logger_->log_debug("Releasing [%p] script engine", engine.get());
+    engine_queue_.enqueue(engine);
```

Comment:
std::move could be used

Reply:
Updated in 268d10998075d418fa7fca9fbd3d318996bda6bd